File: kotlin/lib/library/src/main/java/com/xplpc/util/Log.kt
package com.xplpc.util

import android.util.Log
import com.xplpc.library.BuildConfig
import com.xplpc.system.Constants

object Log {
    val ENABLED: Boolean = BuildConfig.DEBUG

    fun d(msg: String) {
        if (ENABLED) {
            Log.d(Constants.LOG_GROUP, msg)
        }
    }

    fun e(msg: String) {
        Log.e(Constants.LOG_GROUP, msg)
    }
}

File: swift/lib/Sources/Util/Log.swift
import Foundation
import os.log

struct Log {
    /// The library writes through a subsystem of its own, so a consumer can filter or silence it and can raise it to debug without a rebuild.
    private static let log = OSLog(subsystem: "XPLPC", category: "xplpc")

    static func d(_ msg: String) {
        os_log("%@", log: log, type: .debug, msg)
    }

    static func e(_ msg: String) {
        os_log("%@", log: log, type: .error, msg)
    }
}
